contracts/Escrow.sol

```diff
@@ -0,0 +1,171 @@
+    */
+    function _getLatestCollateralRatio() internal returns(uint) {
+        uint debtValue = ILoan(loan).accrueInterest();
+        uint collateralValue = _getCollateralValue();
+        if(collateralValue == 0) return 0;
+        if(debtValue == 0) return MAX_INT;
+
+        return _percent(collateralValue, debtValue, 18);
+    }
+
+    // https://stackoverflow.com/questions/42738640/division-in-ethereum-solidity/42739843#42739843
+    function _percent(uint numerator, uint denominator, uint precision) internal pure returns(uint quotient) {
+        uint _numerator  = numerator * 10 ** (precision + 1);
+        // with rounding of last digit
+        uint _quotient =  ((_numerator / denominator) + 5) / 10;
+        return ( _quotient);
+    }
+
+    /*
+    * @dev calculate the USD value of the collateral stored
+    * @returns - the collateral's USD value
+    */
+    function _getCollateralValue() internal returns(uint) {
+        uint collateralValue = 0;
+        for(uint i = 0; i < _tokensUsedAsCollateral.length; i++) {
+            address token = _tokensUsedAsCollateral[i];
+            uint deposit = deposited[token];
+            if(deposit != 0) {
+                uint price = IOracle(oracle).getLatestAnswer(token);
+                // need to scale value by token decimal
+                (bool success, bytes memory result) = token.call(abi.encodeWithSignature("decimals()"));
+                if(!success) {
+                    collateralValue += price * (deposit / 1e18);
+                } else {
+                    uint8 decimals = abi.decode(result, (uint8));
+                    collateralValue += price * (deposit / (1 * 10 ** decimals));
+                }
+            }
+        }
+
+        return collateralValue;
+    }
+
+    /*
+    * @dev see IEscrow.sol
+    */
+    function addCollateral(uint amount, address token) external returns(uint) {
+        require(amount > 0, "Escrow: amount is 0");
+        require(
+            IOracle(oracle).getLatestAnswer(token) != 0,
+            "Escrow: deposited token does not have a price feed"
+        );
+        require(IERC20(token).transferFrom(msg.sender, address(this), amount));
+        deposited[token] += amount;
+        _addTokenUsed(token);
+        emit CollateralAdded(token, amount);
+
+        return _getLatestCollateralRatio();
+    }
+
+    /*
+    * @dev track the tokens used as collateral
+    */
+    function _addTokenUsed(address token) internal {
+        if(!_tokensUsed[token]) {
+            _tokensUsed[token] = true;
+            _tokensUsedAsCollateral.push(token);
+        }
+    }
+
+    /*
+    * @dev see IEscrow.sol
+    */
+    function releaseCollateral(uint amount, address token, address to) external returns(uint) {
+        require(amount > 0, "Escrow: amount is 0");
+        require(msg.sender == borrower, "Escrow: only borrower can call");
+        require(IERC20(token).transfer(to, amount));
+        deposited[token] -= amount;
```

Comment:
This is potential reentrancy bug. Can you move balance update before token transfer and add another check like `require(amount <= deposited[token]`. 

Same thing in the `liquidate` function too please but its less of an issue since thats controlled by loan smart contract not an independent actor.